isisd/isis_spf.c

```diff
@@ -1190,14 +1236,46 @@ static void add_to_paths(struct isis_spftree *spftree,
 		   vertex->d_N);
 #endif /* EXTREME_DEBUG */
 
+	if (VTYPE_IS(vertex->type)
+	    && !CHECK_FLAG(spftree->flags, F_SPFTREE_NO_ADJACENCIES)) {
+		if (listcount(vertex->Adj_N) > 0) {
+			if (spftree->type == SPF_TYPE_TI_LFA) {
+				struct isis_adjacency *adj;
+
+				if (isis_lfa_check(spftree, vertex) != 0)
+					return;
+
+				adj = isis_adj_find(area, spftree->level,
+						    vertex->N.id);
+				if (adj)
+					sr_adj_sid_add_single(
+						adj, spftree->family, true,
+						vertex->Adj_N);
+			}
+		} else if (IS_DEBUG_SPF_EVENTS)
+			zlog_debug(
+				"ISIS-Spf: no adjacencies do not install backup Adj-SID for %s depth %d dist %d",
+				vid2string(vertex, buff, sizeof(buff)),
+				vertex->depth, vertex->d_N);
+	}
+
 	if (VTYPE_IP(vertex->type)
 	    && !CHECK_FLAG(spftree->flags, F_SPFTREE_NO_ROUTES)) {
-		if (listcount(vertex->Adj_N) > 0)
+		if (listcount(vertex->Adj_N) > 0) {
+			struct route_table *route_table;
+
+			if (spftree->type == SPF_TYPE_TI_LFA) {
+				if (isis_lfa_check(spftree, vertex) != 0)
+					return;
+				route_table = spftree->lfa.old.spftree
+						      ->route_table_backup;
+			} else
+				route_table = spftree->route_table;
+
 			isis_route_create(&vertex->N.ip.dest, &vertex->N.ip.src,
 					  vertex->d_N, vertex->depth,
-					  vertex->Adj_N, spftree->area,
-					  spftree->route_table);
-		else if (IS_DEBUG_SPF_EVENTS)
+					  vertex->Adj_N, area, route_table);
+		} else if (IS_DEBUG_SPF_EVENTS)
 			zlog_debug(
 				"ISIS-Spf: no adjacencies do not install route for %s depth %d dist %d",
```

Comment:
nit -- I know this is in the original code, but the same nit as above ... :-)

Reply:
Yeah I fixed that as well :)